indexer/packages/postgres/src/models/affiliate-info-model.ts

```diff
@@ -0,0 +1,76 @@
+import { NonNegativeNumericPattern } from '../lib/validators';
+import UpsertQueryBuilder from '../query-builders/upsert';
+import BaseModel from './base-model';
+
+export default class AffiliateInfoModel extends BaseModel {
+  static get tableName() {
+    return 'affiliate_info';
+  }
+
+  static get idColumn() {
+    return 'address';
+  }
+
+  static get jsonSchema() {
+    return {
+      type: 'object',
+      required: [
+        'address',
+        'affiliateEarnings',
+        'referredMakerTrades',
+        'referredTakerTrades',
+        'totalReferredFees',
+        'totalReferredUsers',
+        'referredNetProtocolEarnings',
+        'firstReferralBlockHeight',
+      ],
+      properties: {
+        address: { type: 'string' },
+        affiliateEarnings: { type: 'int', pattern: NonNegativeNumericPattern },
+        referredMakerTrades: { type: 'int', pattern: NonNegativeNumericPattern },
+        referredTakerTrades: { type: 'int', pattern: NonNegativeNumericPattern },
+        totalReferredFees: { type: 'int', pattern: NonNegativeNumericPattern },
+        totalReferredUsers: { type: 'int', pattern: NonNegativeNumericPattern },
+        referredNetProtocolEarnings: { type: 'int', pattern: NonNegativeNumericPattern },
+        firstReferralBlockHeight: { type: 'int', pattern: NonNegativeNumericPattern },
```

Comment:
![CleanShot 2024-08-30 at 21 58 46@2x](https://github.com/user-attachments/assets/a61c7319-7627-467d-be91-54815f33ccdc)
Only the commented values should be NonNegativeNumericPattern, and I believe the patterns have only been used with strings, I don't know if they can be used with an int